### ISSUE TITLE
Docker auto-updates and breaks Localdev

### ISSUE THREAD

**Issue by conantp** [pantheon-systems/localdev-issues#106]
MacOS Big Sir 11.3.1 / LocalDev 1.1.0

When I first open LocalDev, it installs a compatible version of Docker, which is great. 

However, Docker now appears to auto-update, so when I quit Localdev, Docker updates itself. 

The next time I open LocalDev, the new (3.3.3) version of Docker starts and updates the configuration files. 

LocalDev will then reinstall an older version of LocalDev, which then throws a Docker error that the configuration file is incompatible. I tell Docker to "Restore to Factory Defaults," and it hangs. 

To resolve the issue, I have to manually update Docker to the latest version (3.3.3), manually uninstall Docker, then open LocalDev and let it install the older version of Docker again. It makes LocalDev basically unusable for me.

**Comment by tarnus** [comment]
Having the exact issue.   Is there a solution yet?

**Comment by erilot** [comment]
Yeah this is causing major pain here too. It looks like this is breaking a lot of things (see [this thread](https://github.com/docker/roadmap/issues/183)). Right now I have an urgent fix that needs to go up, but I can't use localdev at all and fixing the docker installation will take quite a long time... so I'm falling back to SFTP hackery on a multidev environment.

Docker appears to have stopped forcing the auto-update, but until LocalDev is on the latest version of Docker... we are in this cycle.

**Comment by girlg33k** [comment]
I'm having the same issue, extremely frustrating.

**Comment by matthewboman** [comment]
I was able to get it running by deleting Docker, running `rm -r ~/Library/Group\ Container/groups.docker.com`, and then letting Localdev re-install it. Not a permanent solution but a temporary fix.

**Comment by mIIwaukee** [comment]
> I was able to get it running by deleting Docker, running `rm -r ~/Library/Group\ Container/groups.docker.com`, and then letting Localdev re-install it. Not a permanent solution but a temporary fix.

I came here to report that this also works for me. Not ideal but a good quick fix.

**Comment by steviewanders** [comment]
Is this fix still working for people? Could somebody describe their working setup/versions please?

Even when I remove all Docker and let Localdev install it's 2 or 3.x or whatever version, it is reliably broken in a loop showing this error dialog.

 
<img width="1420" alt="Screen Shot 2021-09-01 at 3 46 43 PM" src="https://user-images.githubusercontent.com/71902971/131755367-cc105b39-f253-4f40-8849-2c9a1a22d526.png">

**Comment by steviewanders** [comment]
A fix for my problem above: 

1) Uninstall Docker Desktop and Localdev completely, enjoy the catharsis. 

2) Install Docker Desktop 3.1.0 from https://docs.docker.com/desktop/mac/release-notes/#docker-desktop-310. 

2) Install Localdev and when it first mentions the Docker version incompatibility you need to select the option that allows you "Unsupported versions" - or if your Localdev is working... goto Settings and select that option.

<img width="1420" alt="Screen Shot 2021-09-01 at 5 00 45 PM" src="https://user-images.githubusercontent.com/71902971/131760717-ae574d9f-1c3f-455c-884b-64f21f7662c2.png">


Alternatively Lando is being recommended by Pantheon people in their Slack #localdev channel so perhaps save yourself this process :)

**Comment by stazna01** [comment]
Right after installing the Docker version with Localdev, immediately

Go into Terminal (change $USER to your username on the machine):

- cd /Users/$USER/Library/Caches/com.docker.docker/org.sparkle-project.Sparkle
- Delete anything in this folder (it likely already has downloaded the new Docker version which would install on restart)

Switch to root user:

- sudo su

Change directory ownership and prevent access to other users (change $USER to your username on the machine):

- cd /Users/$USER/Library/Caches/com.docker.docker/
- chown -R root org.sparkle-project.Sparkle
- chmod -R 700 org.sparkle-project.Sparkle

Now Docker won't have access to download updates since it tries to put them in the folder it can no longer write to.

Slightly adapted instructions from: https://gist.github.com/LeZuse/aad7d02d2ccc564f7fc21f032607b47b

**Comment by tbriggs-mx** [comment]
As a developer who works with several different environments outside of LocalDev, it's really frustrating that I'm stuck on an outdated version of Docker. I'm also noticing that on macOS Monterey, when I'm running LocalDev, it really bogs down my computer. I have bumbed up the resources in Docker quite a bit, but I'm not seeing much of a difference.
